tests/integration/main-test.js

```diff
@@ -58,6 +58,57 @@ module('Integration | main', function(hooks) {
 
     assert.equal(user.get('profile.firstName'), 'Terry');
     assert.equal(user.get('profile.lastName'), 'Bubblewinkles');
+
+    changeset.set('profile', null);
+    assert.equal(changeset.get('profile'), null, 'changeset profile is null');
+
+    changeset.execute();
+
+    assert.equal(changeset.get('profile'), null, 'changeset profile relationship is still null');
+    assert.equal(user.get('profile').get('firstName'), null, 'underlying user profile firstName is null');
+    assert.ok(user.get('profile'), 'user has yet to call save so still present as proxy');
```

Comment:
@nickshot  Lmk if this is what you expect based on the tests.  I believe we could close #315 but wanted to check.